src/parser/mod.rs

```diff
@@ -4139,6 +4160,167 @@ impl<'a> Parser<'a> {
         })
     }
 
+    pub fn parse_create_trigger(&mut self, or_replace: bool) -> Result<Statement, ParserError> {
+        if dialect_of!(self is PostgreSqlDialect) {
+            let name = self.parse_object_name(false)?;
+            let period = self.parse_trigger_period()?;
+
+            let event = self.parse_keyword_separated(Keyword::OR, Parser::parse_trigger_event)?;
+            let table_name = if self.parse_keyword(Keyword::ON) {
+                self.parse_object_name(false)?
+            } else {
+                return self.expected("keyword `ON`", self.peek_token());
+            };
+
+            let mut referencing = vec![];
+            if self.parse_keyword(Keyword::REFERENCING) {
+                while let Some(refer) = self.parse_trigger_referencing()? {
+                    referencing.push(refer);
+                }
+            }
+
+            let (for_each, include_each) = if self.parse_keywords(&[Keyword::FOR]) {
+                let include_each = self.parse_keyword(Keyword::EACH);
+                (
+                    match self.parse_one_of_keywords(&[Keyword::ROW, Keyword::STATEMENT]) {
+                        Some(Keyword::ROW) => Some(TriggerObject::Row),
+                        Some(Keyword::STATEMENT) => Some(TriggerObject::Statement),
+                        _ => {
+                            return self.expected("an `ROW` OR `STATEMENT`", self.peek_token());
+                        }
+                    },
+                    include_each,
+                )
+            } else {
+                (None, false)
+            };
+
+            let condition = if self.parse_keyword(Keyword::WHEN) {
+                Some(self.parse_expr()?)
+            } else {
+                None
+            };
+
+            let exec_body = if self.parse_keyword(Keyword::EXECUTE) {
```

Comment:
similar to the one_of, we also have a [self.expect_keyword](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/parser/mod.rs#L3317)

(can we do a pass through the changes to replace the `if parse .. else error` pattern to reuse the functionality?)

Reply:
Thank you for pointing out the `expect_keyboard` method, I have refactored the code accordingly now in every position where I found a similar pattern.